cabal-install/src/Distribution/Client/CmdListBin.hs

```diff
@@ -123,12 +123,13 @@ listbinAction flags@NixStyleFlags{..} args globalFlags = do
         Nothing  -> die' verbosity "No or multiple targets given..."
         Just gpp -> return $ IP.foldPlanPackage
             (const []) -- IPI don't have executables
-            (elaboratedPackage distDirLayout (elaboratedShared buildCtx))
+            (elaboratedPackage distDirLayout (elaboratedShared buildCtx) selectedComponent)
             gpp
 
     case binfiles of
+        []     -> die' verbosity "No target found"
         [exe] -> putStrLn exe
-        _     -> die' verbosity "No or multiple targets given"
+        execs -> die' verbosity $ "Multiple targets found: " <> show execs
```

Comment:
nitpick: s/execs/exes/ for consistency with the rest of the codebase

Reply:
oh, and rather than `show` please use `prettyShow`

Reply:
`prettyShow` didn't work because of a missing `Pretty` instance for `[FilePath]`. I just remove the part the shows the alternatives from the message, which seems to be consistent with the remainder of the UI of the `list-bin` command.